src/renderers/webgl/WebGLBindingStates.js

```diff
@@ -379,7 +384,7 @@ function WebGLBindingStates( gl, extensions, attributes, capabilities ) {
 
 					// check for integer attributes (WebGL 2 only)
 
-					const integer = ( capabilities.isWebGL2 === true && ( type === gl.INT || type === gl.UNSIGNED_INT || geometryAttribute.gpuType === IntType ) );
+					const integer = ( capabilities.isWebGL2 === true && ( geometryAttribute.gpuType === ByteType || geometryAttribute.gpuType === ShortType || geometryAttribute.gpuType === IntType || geometryAttribute.gpuType === UnsignedByteType || geometryAttribute.gpuType === UnsignedShortType || geometryAttribute.gpuType === UnsignedIntType ) );
```

Comment:
Previously we'd discussed `gpuType` only being able to be set to `IntType` and `FloatType` since the native type can be inferred from there based on the typed array value (see https://github.com/mrdoob/three.js/pull/21606#issuecomment-1534759990)

Reply:
I see that supersedes your suggestion of `BufferAttribute.integer` which I mentioned in https://github.com/mrdoob/three.js/pull/26076#discussion_r1197272061. Thanks for the reference, I didn't follow the conversation before.